### DISCUSSION TITLE
What features would you like to see?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#8] [Ask a question] [answered]
Hi! We made Yellow and it would be great to collect more ideas. What would you like to see? Anything not working or missing that you need?

**Comment by ofisia** [comment]
An option to edit (the default) username and password in the admin menu be very helpful for non command line-savvy users like me. :smiley:

**Comment by nibreh** [comment]
Hi - For the blog template : maybe create new post blog (pages) directly in the browser like the entire site (if I'm not mistaken, we can't do that with this template)

(bad english, sorry ..)

**Comment by nibreh** [comment]
Ok - I'm really a noob ! ;) We can create blog page in the browser ...

**Comment by markseuffert** [comment]
Great ideas, thanks for your comments. About user accounts, it would be nice to handle user accounts and forgotten passwords in the browser. Maybe as part of the installation or login? About the blog, I'll add the possibility to add and delete pages in the next update. Right now it's too difficult to add pages in the browser, don't think many people will find it.

Thanks, feedback is much appreciated! :)

**Comment by nibreh** [comment]
Yep cool ! And another idea : add possibility to rename a page ?

By the way : great stuff and simple to use, I like it !

**Comment by ofisia** [comment]
A plugin which output media(s) from each page's folder, just like [Stacey](http://staceyapp.com). It would be very useful for portfolio sites, IMHO.

Example directory:

```
content/2-about/page.txt
content/2-about/page-images/01.jpg
content/2-about/page-images/02.jpg
content/2-about/page-images/03.jpg
```

Example usage:

```
---
Title: About
---

Lorem ipsum dolor sit amet ...

[page-images]

Lorem ipsum dolor sit amet ...
```

**Comment by markseuffert** [comment]
Thanks again for the ideas! About portfolio site, I think it's possible to make an extension that generates output with images or galleries. Do you have an example page how it should look like?

**Comment by nibreh** [comment]
Personnaly, I use fancybox with thumbnail navigation and I create a gallery with table ... 

Here is a my example with yellowcms :
http://nibreh.legtux.org/drafts/yellow

I modified the plugin image.php to get automatically link on picture.jpg and the CSS for a responsive table ... I change the template too, as you can see !

If it gives some ideas ...

**Comment by markseuffert** [comment]
Fancybox plugin sounds like a good idea. :+1:

**Comment by ofisia** [comment]
Another idea: the ability to output **only** the image path (`media/images/picture.jpg`) instead of a well formed `img` tag so it can be used inside [`srcset`](http://responsiveimages.org/) or [`data`](http://html5doctor.com/html5-custom-data-attributes/) attribute.

Since there are already a lot of [gallery/slideshow/lightbox JS](http://bit.ly/U58GpS) out there, perhaps it is better to keep Yellow minimalist by using lists only, for example:

```
<ol>
   <li><img src='01.jpg'></li>
   <li><img src='02.jpg'></li>
   <li><img src='03.jpg'></li>
</ol>
```

From there, users can enhance it to their liking using third party gallery JS above. But, this is just an opinion. :smile:

**Comment by markseuffert** [comment]
Creating lists of media files sounds like a good idea. Let's add it to the list. :+1:

**Comment by nibreh** [comment]
What about a snippet for dropdown menu navigation ?
I know how to do this with HTML and CSS, but not with yellowCMS ...
I'm reading snippets php files for example, and testing some stuff, but I'm lost ;)
I'm curious on how to do this.

**Comment by markseuffert** [comment]
Can I help you with the dropdown menu? If you have an example HTML code, I could write a snippet.

For developers there's a [Yellow API](https://datenstrom.se/yellow/help/api-for-developers), which can be used to build all sorts of website navigations. The default navigation is very basic. But this can be changed with snippets and there are a few examples to start with: navigationsub, navigationtree and breadcrumbs. The possibilities are endless. If you know what the HTML code should look like, one can write a snippet to generate it.

**Comment by nibreh** [comment]
Thanks markseu. I will experiment with Yellow API to learn some stuff. First I must translate from english to understand ;)

The HTML should look like this with yellow CMS :

```
<div class="navigation"> 
 <ul>
  <li class="active"><a href="/yellow">Home</a></li>

  <li><a href="#">About (The Folder)</a>  
    <ul>
    <li class="active"><a href="/yellow/about/page-1">page-1</a></li>
    <li><a href="/yellow/about/page-2">page-2</a></li>
    <li><a href="/yellow/about/page-3">page-3</a></li>
    </ul>
 </li>

    <li><a href="/yellow/blog">Blog</a></li>
    <li><a href="/yellow/wiki">Wiki</a></li>
  </ul>
</div>
```

Or something like that. CSS do the rest - 

page.txt (index page for the folder About) is useless whith this method ... ?

I would like something like that for a friend, it's a simple mobile navigation by webdesignerwall :
http://webdesignerwall.com/demo/mobile-nav/

cheers - (maybe I'm wrong somewhere)

**Comment by markseuffert** [comment]
Have you tried the navigationtree snippet? It generates HTML code similar to your example.

**Comment by nibreh** [comment]
Yes of course, but it's not exactly the same code

Short version :
`<ul><li><ul></ul></li></ul>` in my example  
`<ul><li></li><ul><li></li></ul></ul>` with navigationtree

CSS must be different of course, but I don't know if its possible with this structure. I will try anyway, it's a good way to learn

```
<ul>
<li><a class="active" href="/site/yellowcms/">Home</a></li>

<li><a href="/site/yellowcms/about/">about</a></li>
<ul>
<li><a href="/site/yellowcms/about/page-1">page-1</a></li>
<li><a href="/site/yellowcms/help/page-2">page-2</a></li>
</ul>

</ul>
```

**Comment by markseuffert** [comment]
Yes, you are absolutely right! A nested list should be inside the `<li>` element.
I'll fix the snippet and add example CSS for testing. Thanks for the explanation! :smiley:

**Comment by nibreh** [comment]
NEW FEATURE / IDEA : 

An historic with old saved pages for the Wiki, would be nice for multiusers

Maybe I asked too much ;)

For now I'm enjoy Yellow, I'm ready to use it for my little and personnal website

**Comment by markseuffert** [comment]
Thanks for the idea! :smiley:

You can use Dropbox to sync your website. It has a history of changes. Not every web hosting supports it, you might want to check with your web hosting company. If you have your own Linux server, see https://www.dropbox.com/install?os=lnx

**Comment by markseuffert** [comment]
@junibreh: I updated the navigationtree snippet. Now it should be possible to create any type of menu, drop down, vertical, etc. Hope this helps.

**Comment by dajare** [comment]
@markseu - Would it be possible to have the "Author name" (now in `config.ini`) picked up from the User information? That way it would be possible to have multiple users on a site, with posts identified by the user that posted it. This would make Yellow a great platform for a "team blog", etc.

Thanks! (Great piece of software, IMO!) :+1:

**Comment by markseuffert** [comment]
@dajare Thanks for testing Yellow and giving feedback, much appreciated. You can specify the author on every page. Unfortunately this is not done automatically when creating a new page. The settings at the top of `config.ini` are defaults and can be overwritten in the settings of a page. Here's an example:

```
---
Title: Blog posting
Published: 2014-11-12
Author: David
Tag: Example
---
Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor
incididunt ut labore et dolore magna pizza. Ut enim ad minim veniam, 
quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

It should give you a lot of flexibility, multiple authors and tags are possible too. Does this work for you?

**Comment by dajare** [comment]
> It should give you a lot of flexibility, multiple authors and tags are possible too. Does this work for you?

Yes, it does -- and I should have noticed that! It might be a small enhancement to consider picking up the author from the `user` data, but using the _page_ setting is simple and effective.

Thanks for the reply!

**Comment by markseuffert** [comment]
Setting author when creating a page is a good idea! Let's add it to the TODO list. :+1:

**Comment by markseuffert** [comment]
I updated the blog. Now it's setting current date and author. Thanks for the idea David, very useful!

**Comment by dajare** [comment]
Thanks for that!

Here's another suggestion - how about adding the [PHP Smartypants](https://github.com/michelf/php-smartypants) extension to MarkdownExtra? It's the "finishing touch" on the filter (IMO). There's also the "Typographer" extension to Smartypants that adds some further configuration possibilities (esp. for different locales).

I _think_ this is a fairly trivial addition. See what you think!

**Comment by markseuffert** [comment]
Yes it looks easy to integrate, e.g. as a plugin, I'll have a closer look after next update.

Thanks for the idea. Anything else missing, please let me/us know. :smiley:

**Comment by markseuffert** [comment]
I added a Smartypants plugin. Hope you like it, @dajare.

**Comment by dajare** [comment]
@markseu I like it, a lot! :+1: :tada:  I hope others do, too. Thanks!

**Comment by markseuffert** [comment]
I added an image list snippet. Please don't expect too much. It's a first step towards an image gallery.

**Comment by markseuffert** [comment, marked as answer]
Thanks everyone for their ideas and help, appreciate it! Hope the the most important features are there, I know that many things are still missing, for example image galleries. If you are interested in a feature or need something, please open a new issue or consider [contributing to Yellow](https://datenstrom.se/yellow/help/contributing-guidelines).
